packages/overlay/README.md

```diff
@@ -11,190 +11,274 @@ Overlays in Spectrum Web Components are created via the `Overlay` class system,
+        </sp-action-group>
+    </sp-popover>
+</sp-overlay>
+<sp-overlay ?delayed="${delayed}" trigger="trigger-3@hover">
+    <sp-tooltip>Hover</sp-tooltip>
+</sp-overlay>
+<sp-overlay
+    trigger="trigger-3@longpress"
+    type="auto"
+    placement="right-start"
+    .offset="${popoverOffset}"
+>
+    <sp-popover class="overlay-demo-popover" tip>
+        <sp-action-group vertical quiet>
+            <sp-action-button>
+                <sp-icon-anchor-select slot="icon"></sp-icon-anchor-select>
+            </sp-action-button>
+            <sp-action-button>
+                <sp-icon-polygon-select slot="icon"></sp-icon-polygon-select>
+            </sp-action-button>
+            <sp-action-button>
+                <sp-icon-rect-select slot="icon"></sp-icon-rect-select>
+            </sp-action-button>
+        </sp-action-group>
+    </sp-popover>
+</sp-overlay>
 ```
-type TriggerInteractions =
-    | 'click'
-    | 'custom',
-    | 'hover'
-    | 'inline'
-    | 'modal'
-    | 'replace';
+
+## API
+
+```ts
+<sp-overlay
+    ?open=${boolean}
+    ?delayed=${boolean}
+    offset=${Number | [Number, Number]}
+    placement=${Placement}
+    receives-focus=${'true' | 'false' | 'auto' (default)
+    trigger=${string | ${string}@${string}}
+    .triggerElement=${HTMLElement}
+    .triggerInteraction=${'click' | 'longpress' | 'hover'}
+    type=${'auto' | 'hint' | 'manual' | 'modal' | 'page'}
+></sp-overlay>
 ```
 
-`click` will open an overlay that will close immediately on the next click that is not on an element within the overlay.
+# Events
 
-`custom` is less opinionated and allows for some customization of the process from the outside.
+When fully open the `<sp-overlay>` element will dispatch the `sp-opened` event, and when fully closed the `sp-closed` event will be dispatched. "Fully" in this context means that all CSS transitions that have dispatched `transitionrun` events on the direct children of the `<sp-overlay>` element have successfully dispatched their `transitionend` event. Keep in mind the following:
 
-`hover` will close the overlay as soon as the pointer leaves the trigger to which the overlay is connected.
+-   `transition*` events bubble; this means that while transition events on light DOM content of those direct children will be heard, those events will not be taken into account
+-   `transition*` events are not composed; this means that transition events on shadow DOM content of the direct children will not propagate to a level in the DOM where they can be heard
 
-`inline` places the overlay after the trigger but before the next element in the logical tab order. This means the `shift + tab` keyboard stroke will return to the trigger.
+This means that in both cases, if the transition is meant to be a part of the opening or closing of the overlay in question you will need to redispatch the `transitionrun` and `transitionend` events from that transition from the closest direct child of the `<sp-overlay>`.
 
```

Comment:
@hunterloftis, added this section in association with the API changes we discussed this morning for `guaranteedAllTransitionend`.

Reply:
I didn't even consider composition and bubbling, TIL